File: 2018/01/calc-freq.py
#!/usr/bin/env python3
import sys


def main(inputfile):
    with open(inputfile, 'r') as rd:
        print("Resulting frequency:", calculate_freq(0, rd))


def calculate_freq(start, rd):
    mutations = [int(x.strip()) for x in rd.readlines()]
    for mut in mutations:
        start += mut
    return start


if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: %s inputfile", sys.argv[0])
        sys.exit(0)
    main(sys.argv[1])

File: 2018/01/calc-freq.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
)

func main() {
	if len(os.Args) >= 2 {
		calculate_freq(0, os.Args[1])
	} else {
		fmt.Println("Please also give us a file to understand....")
	}
}

func calculate_freq(start int, inputfile string) {
	current := start

	file, err := os.Open(inputfile)
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		i, err := strconv.Atoi(scanner.Text())
		if err != nil {
			log.Fatal(err)
		}
		current = current + i
	}

	if err := scanner.Err(); err != nil {
		log.Fatal(err)
	}

	fmt.Println("Resulting frequency:", current)
}
